src/main/java/com/hcl/appscan/sdk/utils/ServiceUtil.java

```
@@ -223,6 +223,47 @@ private static boolean hasEntitlement(String scanType, IAuthenticationProvider p
 		return false;
 	}
 
+	/**
+     * Checks if the given scanId is valid for scanning.
+     *
+     * @param scanId The scanId to test.
+     * @param applicationId The applicationId to verify.
+     * @param type The scanType to verify.
+     * @param provider The IAuthenticationProvider for authentication.
+     * @return True if the scanId is valid. False is returned if the scanId is not valid, the request fails, or an exception occurs.
+     */
+	public static boolean isScanId(String scanId, String applicationId, String type, IAuthenticationProvider provider) {
+        if (provider.isTokenExpired()) {
+            return true;
+        }
+
+        String request_url = provider.getServer() + API_BASIC_DETAILS;
+        request_url += "?$filter=Id%20eq%20" + scanId + "&%24select=AppId%2C%20Technology";
+        Map<String, String> request_headers = provider.getAuthorizationHeader(true);
+
+        HttpClient client = new HttpClient(provider.getProxy(), provider.getacceptInvalidCerts());
+        try {
+            HttpResponse response = client.get(request_url, request_headers, null);
+
+            if (response.isSuccess()) {
+                JSONObject obj = (JSONObject) response.getResponseBodyAsJSON();
+                JSONArray array = (JSONArray) obj.get(ITEMS);
+                if (array.isEmpty()) {
+                    return false;
+                } else {
+                    JSONObject body = (JSONObject) array.getJSONObject(0);
+                    String appId = body.getString(CoreConstants.APP_ID);
+                    String technologyName = body.getString("Technology");
+                    return appId.equals(applicationId) && technologyName.equals(updatedScanType(type));
+                }
+            }
+        } catch (IOException | JSONException e) {
```

Rather than catching and ignoring these exceptions, it would be better to throw them and allow the caller to handle the error condition (e.g. by logging something).

---

Okay, Sure. An Error will be thrown in the console logs using the Logger.

---

Since this is an SDK, it would be better to throw the exception, so the consumer of the SDK can handle it however they see fit (error messsage, log, etc.). I suggesting catching the JSONException and throwing an IOException initialized with it.

---

Okay, sure